src/components/signUpInOut/SignUpForm.jsx

```diff
@@ -0,0 +1,149 @@
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import {
+  Form, Button, Row, Col, Modal,
+} from 'react-bootstrap';
+import { FaXmark } from 'react-icons/fa6';
+import { signUp } from '../../features/authSlice';
+
+const initialAddressData = {
+  street: '',
+  city: '',
+  state: '',
+  zip_code: '',
+};
+
+const initialFormData = {
+  name: '',
+  age: '',
+  email: '',
+  photo: '',
+  role: '',
+  password: '',
+  password_confirmation: '',
+  address: { ...initialAddressData },
+};
+
+const SignUpForm = ({ showSignUpModal, handleCloseSignUpModal, handleOpenSignUpModal }) => {
+  const [formData, setFormData] = useState({ ...initialFormData });
+  const dispatch = useDispatch();
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setFormData((prevFormData) => {
+      if (name.startsWith('address.')) {
+        const addressField = name.split('.')[1];
+        return {
+          ...prevFormData,
+          address: {
+            ...prevFormData.address,
+            [addressField]: value,
+          },
+        };
+      }
+      return {
+        ...prevFormData,
+        [name]: value,
+      };
+    });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    dispatch(signUp(formData));
+    window.location.href = '/';
```

Comment:
- [ ] The only way I could find to correctly create a user (locally) is to delete this line `window.location.href = '/';`.

As said above, This is because when the submit button is clicked, this line immediately goes to "/", making another sudden request to the server before it saves the user.

When I remove that line, it works again:

![image](https://github.com/baqar-abbas/appointment-app-front-end/assets/57421823/e592b13d-36cb-4f42-8344-43678abe571a)


- [ ] In the live demo (after waiting for the delay), if I try to create a user and login this happens:

![image](https://github.com/baqar-abbas/appointment-app-front-end/assets/57421823/ee77ac63-789c-4bdc-ae3b-1b4935a79ac8)

_note: Both issues were tested on Chrome and Firefox.  On 2 computers.  The live demo was tested on Windows and Linux using both browsers._


- [ ] There is functionality to sign in as a doctor, but throws this error:

![image](https://github.com/baqar-abbas/appointment-app-front-end/assets/57421823/8cb64949-58ee-4be0-9eb0-ddb81101a429)